app/Http/Controllers/Auth/LoginController.php

```diff
@@ -68,70 +68,118 @@ protected function verifyLogin(Request $request) {
-                                $opt = new Opt;
-                                $opt->controller_id = $res['cid'];
-                                $opt->ip_address = '0.0.0.0';
-                                $opt->means = 'VATUSA API';
-                                $opt->option = 0;
-                                $opt->save();
-                                $userstatuscheck->opt = 0;
-                            }
-                        }
-                        if ($userstatuscheck->visitor == '1') {
-                            if ($res['facility'] != 'ZZN') {
-                                $userstatuscheck->visitor_from = $res['facility'];
-                            }
-                        } else {
-                            $userstatuscheck->visitor_from = null;
-                        }
-                        $userstatuscheck->save();
-                        Auth::loginUsingId($userstatuscheck->id, true);
-                    } else { // User was found on the roster, but is not an active home or visiting controller
-                        return redirect('/')->with('error', 'You have not been found on the roster. If you have recently joined, please allow up to an hour for the roster to update.');
-                    }
-                } else { // User was not found on the facility roster as a home or visiting controller
-                    return redirect('/')->with('error', 'You have not been found on the roster. If you have recently joined, please allow up to an hour for the roster to update.');
-                }
-            } else { // VATUSA API responded in an unexpected format (non-JSON)
-                return redirect('/')->with('error', 'We are unable to verify your access at this time. Please try again in a few minutes.');
-            }
-            if ($userstatuscheck->status == 0) { // User is authenticated and has been logged in. LOA is no longer used at ZTL, this item retained for compatibility
-                return redirect('/dashboard')->with('success', 'You have been logged in successfully. Please note that you are on an LOA and should not control until off the LOA. If this is an error, please let the DATM know.');
-            } else { // User is authenticated and has been logged in
-                return redirect()->intended('/dashboard')->with('success', 'You have been logged in successfully.');
-            }
-        } else { // VATUSA API does not respond or throws an error
+
+        if (! $result) {
+            return redirect('/')->with('error', 'We are unable to verify your access at this time. Please try again in a few minutes.');
+        }
+
+        $resu = json_decode($result->getBody()->__toString(), true);
+
+        if (! isset($resu['data'])) {
             return redirect('/')->with('error', 'We are unable to verify your access at this time. Please try again in a few minutes.');
         }
+
```

Comment:
I see what you meant about refactoring this now... looks much better using returns to exit and break up the code versus the complex conditional structure

Reply:
Yeah much more readable. Nesting just over-complicates things: https://medium.com/codex/why-you-shouldnt-nest-your-code-185cf2e2cde3